src/common/rwalk.f90

```diff
@@ -99,18 +107,18 @@ subroutine rwalk(blfullmix,part,pextra)
   if (part%z <= part%tbl) then ! Above boundary layer
       part%z = part%z + vrdbla*rnd(3)
   else ! In boundary layer
-    if (blfullmix) then
-      part%z = 1.0 - (1.0 - part%tbl)*1.1*(rnd(3)+0.5)
-    else ! not full mixing
-      hfactor = part%hbl/hmax
-      rv = lmax*hfactor*tfactor
-      rvmax = 1.0 - part%tbl
-
-      rv = min(rv, rvmax)
+    if (blfullmix .or. (tsqrtfactor_v .gt. 1.0)) then ! full mixing
+      part%z = 1.0 - (1.0 - part%tbl)*(1.+entrainment)*(rnd(3)+0.5)
```

Comment:
Move factor `(1.0 - part%tbl)*(1.+entrainment)` outside if and give a name to it (top_of_entrainment_zone?). Do you need to check if negative as on l.120?

Reply:
Done